### PR TITLE
Add support for dev-scripts and VA1 to reproducer

### PR DESCRIPTION
**Pull request by cjeanner**
This patch introduces support for dev-scripts, a way to deploy a 3-nodes
OCP, in the reproducer.
For now, we can *either* run the standard libvirt_manager to get the CI
layout, *or* run devscripts and end with a 3-OCP + 3-compute layout.

As a pull request owner and reviewers, I checked that:
- [ ] Appropriate testing is done and actually running
- [ ] Appropriate documentation exists and/or is up-to-date:
  - [ ] README in the role
  - [ ] Content of the docs/source is reflecting the changes

### CODE REVIEW

File: ci_framework/roles/reproducer/tasks/epel.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  when:
+    - ansible_facts['distribution'] == 'RedHat'
+  become: true
+  vars:
+    arch: "{{ ansible_facts['architecture'] }}"
+  ansible.builtin.command:
+    cmd: >-
+      subscription-manager repos
+      --enable codeready-builder-for-rhel-9-{{ arch }}-rpms
+
```
Comment (raukadah) [inline ci_framework/roles/reproducer/tasks/epel.yml:19]: It fails on RHEL machine if the system is not subscribed to subscription manager.
```
ASK [reproducer : Enable CRB repositories (RHEL) _raw_params=subscription-manager repos --enable codeready-builder-for-rhel-9-{{ arch }}-rpms] **************
Friday 20 October 2023  07:46:16 +0530 (0:00:00.023)       0:02:55.146 ******** 
fatal: [baremetal]: FAILED! => {"changed": true, "cmd": ["subscription-manager", "repos", "--enable", "codeready-builder-for-rhel-9-x86_64-rpms"], "delta": "0:00:00.243552", "end": "2023-10-20 05:16:21.881923", "msg": "non-zero return code", "rc": 1, "start": "2023-10-20 05:16:21.638371", "stderr": "", "stderr_lines": [], "stdout": "This system has no repositories available through subscriptions.", "stdout_lines": ["This system has no repositories available through subscriptions."]}
```
In CI, we never subscribe to subscription manager, we use the repos populated by rhos-release tool. If there is a missing repo, We add it.

In Devscript role we have a check regarding subscription manager. https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/devscripts/tasks/sub_tasks/11_rhsm.yml#L27

We can add the same here also or reuse the playbook.

The Devscripts role also provides option to pass custom crb repo: https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/devscripts/tasks/sub_tasks/12_yum_repo.yml#L21

I think we can use both of them here. What do you say?

**Review by raukadah — COMMENTED** [review]
Thank you @cjeanner for adding this wonderful integration. I got a working OCP environment deployed. Kudos to you.:-)

I am using `cifmw_devscripts_user: zuul` and also running the ansible-playbook as a zuul user on the hypervisor. Is it ok to use normal user or I need to run reproducer playbook as a root user?

Reason I am asking this during EDPM compute node VM start task I faced the permission error, It might be faced by other users .
```
TASK [libvirt_manager : Start VMs for type compute state=running, name=cifmw-{{ vm_type }}-{{ vm_id }}, uri=qemu:///system] **********************************
Friday 20 October 2023  09:06:47 +0530 (0:00:26.704)       0:52:12.040 ******** 
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: libvirt.libvirtError: Cannot access storage file '/home/zuul/ci-framework-data/workload/compute-0.qcow2' (as uid:107, gid:107): Permission denied
failed: [baremetal] (item=compute-0) => {"ansible_index_var": "vm_id", "ansible_loop_var": "item", "changed": false, "item": 0, "msg": "Cannot access storage file '/home/zuul/ci-framework-data/workload/compute-0.qcow2' (as uid:107, gid:107): Permission denied", "vm_id": 0}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: libvirt.libvirtError: Cannot access storage file '/home/zuul/ci-framework-data/workload/compute-1.qcow2' (as uid:107, gid:107): Permission denied
failed: [baremetal] (item=compute-1) => {"ansible_index_var": "vm_id", "ansible_loop_var": "item", "changed": false, "item": 1, "msg": "Cannot access storage file '/home/zuul/ci-framework-data/workload/compute-1.qcow2' (as uid:107, gid:107): Permission denied", "vm_id": 1}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: libvirt.libvirtError: Cannot access storage file '/home/zuul/ci-framework-data/workload/compute-2.qcow2' (as uid:107, gid:107): Permission denied
failed: [baremetal] (item=compute-2) => {"ansible_index_var": "vm_id", "ansible_loop_var": "item", "changed": false, "item": 2, "msg": "Cannot access storage file '/home/zuul/ci-framework-data/workload/compute-2.qcow2' (as uid:107, gid:107): Permission denied", "vm_id": 2}
```
and
```
[zuul@titan161 logs]$ sudo virsh list --all
 Id   Name              State
----------------------------------
 5    ocp_master_1      running
 6    ocp_master_0      running
 7    ocp_master_2      running
 -    cifmw-compute-0   shut off
 -    cifmw-compute-1   shut off
 -    cifmw-compute-2   shut off

[zuul@titan161 logs]$ groups zuul
zuul : zuul libvirt
[zuul@titan161 logs]$ 
```
Rest looks good, thank you :-)

**Comment by raukadah** [comment]
@cjeanner I am not sure how much this issue useful for you. But adding it here.
I am running the reproducer playbook as a zuul user and used `cifmw_devscripts_user: virtuser` in varfile and then during clone i hit following issue.
```
TASK [devscripts : Clone the dev-scripts repository. repo={{ cifmw_devscripts_repo }}, dest={{ cifmw_devscripts_repo_dir }}, depth=1, single_branch=True, force=True, version=HEAD] ***
Friday 20 October 2023  08:04:29 +0530 (0:00:00.042)       0:06:02.743 ******** 
FAILED - RETRYING: [baremetal]: Clone the dev-scripts repository. (3 retries left).
FAILED - RETRYING: [baremetal]: Clone the dev-scripts repository. (2 retries left).
FAILED - RETRYING: [baremetal]: Clone the dev-scripts repository. (1 retries left).
fatal: [baremetal]: FAILED! => {"attempts": 3, "changed": false, "cmd": "/usr/bin/git clone --origin origin --depth 1 --single-branch https://github.com/openshift-metal3/dev-scripts.git /home/zuul/src/github.com/openshift-metal3/dev-scripts", "msg": "Cloning into '/home/zuul/src/github.com/openshift-metal3/dev-scripts'...\n/home/zuul/src/github.com/openshift-metal3/dev-scripts/.git: Permission denied", "rc": 1, "stderr": "Cloning into '/home/zuul/src/github.com/openshift-metal3/dev-scripts'...\n/home/zuul/src/github.com/openshift-metal3/dev-scripts/.git: Permission denied\n", "stderr_lines": ["Cloning into '/home/zuul/src/github.com/openshift-metal3/dev-scripts'...", "/home/zuul/src/github.com/openshift-metal3/dev-scripts/.git: Permission denied"], "stdout": "", "stdout_lines": []}
```
It might be linked with users permission issue. I have workaround the above issue by switching back zuul user. Please have a look, thank you :-)

File: ci_framework/roles/reproducer/tasks/epel.yml (hunk 1 above)
Comment (cjeanner) [reply ci_framework/roles/reproducer/tasks/epel.yml:19]: I'll wait to get access to the tasks files - Pragadee is modifying the devscripts role and will push those tasks elsewhere. I'd rather avoid duplicating code at this stage.

**Comment by dasm** [comment]
@cjeanner what's the status of that branch? It has bunch of conflicts.

**Comment by cjeanner** [comment]
Feature is in now, we can drop that PR.

**Pull request closed without merging**